### PR TITLE
feat: add GitHub repo preview with commits and topics (#189)

### LINKED ISSUE #189: [Feature] Real GitHub API-backed repo preview for resource detail pages
## Summary
Follow-up to #180 (removed the fake hardcoded `RepositoryPreview` widget). This proposes building a real replacement — actual commit history and topics fetched from GitHub for resources that have a `github_url`.

## Background
#180 removed `RepositoryPreview`'s fake hardcoded commits/watch/fork counts/topics, since none of it was backed by real data. `GithubStatsCard` already shows honest stars/forks/open issues/last-commit — but there's no real commit list or topics anywhere on the page anymore.

There's currently no GitHub API integration anywhere in this repo (no `api.github.com` calls, no `octokit`), and `github_stats` itself appears to be static/backend-provided rather than fetched live from GitHub — so this would be new groundwork, not an extension of an existing pattern.

## Decisions (confirmed 2026-08-17)
- **Data source: fetch in this Next.js app, not backend/CMS.** `github_stats` is *not* currently populated by any backend/CMS process — it's hardcoded `null` in both mapper paths (`src/shared/infrastructure/payload-resource-mapper.ts:49` and `src/modules/resources/infrastructure/repositories/cms.ts:80`), and the Payload `Resources` collection has no `github_stats` field at all. An earlier design doc proposed backend-side caching via a periodic job, but it was never built, and `payload-backend` has no cron/scheduled-job infra today. Rather than stand up new backend job infra, fetch commits/topics server-side in this app using `next: { revalidate: ... }` caching, same pattern as `cms.ts`/`payload.ts`.
- **Auth: authenticated PAT.** Unauthenticated 60 req/hr per IP won't hold up under real traffic on a resource-detail page. Use a GitHub PAT as a server-only env var (never exposed to the client), scoped read-only to public repo metadata. This repo already owns a GitHub credential this way — see `GITHUB_CLIENT_ID`/`GITHUB_CLIENT_SECRET` in `payload-backend/src/app/oauth/github/callback/route.ts` — so this isn't new credential-ownership territory.
- **UI: new, adjacent component (not an extension of `GithubStatsCard`).** `GithubStatsCard` is a fixed 4-tile numeric stat grid (stars/forks/issues/last-commit). Commits and topics are list-shaped data, not another stat tile, so forcing them into the same grid would be an awkward fit. Keep `GithubStatsCard` unchanged for the numeric stats; add a sibling component next to it for commits/topics.
- **Roadmap fit: confirmed.** This fits RATQ's roadmap — go ahead.

## Scope
- [x] Decision recorded in this issue's thread on backend-vs-frontend data source (see Decisions above)
- [ ] Server-side fetch of commits/topics for resources with a `github_url`, cached (`next: { revalidate: ... }`, matching the pattern in `cms.ts`/`payload.ts`)
- [ ] Resources without a `github_url`, or not hosted on GitHub, render no repo-preview section — same pattern as `GithubStatsCard`
- [ ] UI: new component alongside `GithubStatsCard` (see Decisions above)

## Out of Scope
- Any resource type without a GitHub repo (documentation/dataset-only resources)
- Changes to `GithubStatsCard`'s existing stars/forks/issues/last-commit behavior

## Scope note
This is for RATQ platform features (catalog, filtering, discovery, developer dashboard, etc.), not for suggesting a new resource to list, those come from RATQ's connected data sources (e.g. the CMS), not manual submissions here.

## Technical Notes
- `src/modules/resources/components/GithubStatsCard.tsx` — existing honest-data pattern for the numeric stats; add a new sibling component for commits/topics, don't extend this one (see Decisions)
- `src/types/resource.ts` — `GithubStats` type would need `commits`/`topics` fields, or a new type alongside it, since this data now lives in this app
- No existing `api.github.com`/`octokit` usage anywhere in this repo to reference — this is the first
- GitHub PAT should be added as a new server-only env var (not `GITHUB_CLIENT_ID`/`GITHUB_CLIENT_SECRET`, those are OAuth login credentials, not for API reads)

## Difficulty
Intermediate — real external API integration and caching, data-source question now resolved (see Decisions).

## Acceptance criteria
- [x] A maintainer confirms this fits RATQ's roadmap before implementation starts
- [x] Open questions in Background are resolved and recorded in this issue's thread before implementation starts

## Contributor Checklist
- [ ] Read [[CONTRIBUTING.md](https://claude.ai/CONTRIBUTING.md)](../CONTRIBUTING.md)
- [ ] Discuss the approach in the issue if uncertain before starting
- [ ] Keep changes focused to this issue's scope
- [ ] Ensure `tsc-check`/lint/tests pass before opening a PR

### PR DESCRIPTION
**Pull request by motantawi**
Closes #189

## Summary
Adds a real GitHub repo preview on resource detail pages: recent commits and topics for resources with a valid `https://github.com/owner/repo` URL.

The preview is fetched server-side in this Next.js app (not the CMS/backend), cached with `next: { revalidate: 300 }`, and uses a server-only `GITHUB_API_TOKEN`. `GithubStatsCard` is unchanged; a new sibling component renders below it. Resources without a GitHub URL, or with a non-github.com URL, show no preview.

## Approach
- Parse `github_url` and fetch commits (`per_page=5`) + topics from the GitHub API
- Native `fetch` only (edge runtime)
- On missing token or API error, return `null` so the page never crashes
- All UI copy goes through i18n (`en.json` / `ar.json`), RTL-safe

## Screenshots
image

## Test plan
- [x] `npm run test:run` — URL parsing and fetch error handling
- [x] `npm run build`
- [ ] Resource with a valid `github_url` + `GITHUB_API_TOKEN` shows commits (and topics if the repo has them)
- [ ] Resource without `github_url` shows no preview
- [ ] Missing token fails silently; page still loads

## Notes
`GITHUB_API_TOKEN` (read-only GitHub PAT) is required in production. Do not expose it as `NEXT_PUBLIC_*`.

## Summary by CodeRabbit

- **New Features**
  - Added GitHub repository previews to resource detail pages.
  - Displays repository topics and recent commits with localized labels and date formatting.
  - Supports direct links to view individual commits.
  - Added English and Arabic translations.

- **Bug Fixes**
  - Invalid, unavailable, or unsupported repository links now gracefully omit the preview.

### CODE REVIEW

File: src/modules/resources/infrastructure/github/fetchGithubRepoPreview.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  try {
+    const [commitsRes, topicsRes] = await Promise.all([
+      fetch(`${GITHUB_API}/repos/${encodedOwner}/${encodedRepo}/commits?per_page=5`, {
+        headers,
+        next: { revalidate: REVALIDATE },
+      }),
+      fetch(`${GITHUB_API}/repos/${encodedOwner}/${encodedRepo}/topics`, {
+        headers,
+        next: { revalidate: REVALIDATE },
+      }),
```
[the remark this answers is not in the document]
Comment (motantawi) [reply src/modules/resources/infrastructure/github/fetchGithubRepoPreview.ts:72]: Good catch. #189 specified `next: { revalidate: 300 }` like `cms.ts`, so this PR follows that scope. I can add `withEdgeCache` for GitHub previews in this PR or a follow-up if maintainers want parity with the resource cache on edge.

[the remark this answers is not in the document]
Comment (motantawi) [reply src/modules/resources/infrastructure/github/fetchGithubRepoPreview.ts:72]: Thanks. I'll defer withEdgeCache for GitHub previews to a follow-up — please open the issue with the cache-key, TTL, and successful-response requirements. This PR stays scoped to #189 as specified.

**Review by abubakr-itqan — APPROVED** [review]
Reviewed the diff end to end - this matches #189's resolved decisions exactly: server-side fetch in this app (not backend), GITHUB_API_TOKEN as a server-only env var, a new sibling component next to GithubStatsCard rather than extending it, strict URL parsing (https://github.com/{owner}/{repo} only), and returns null on missing token/API failure so the page never crashes. Ran tsc (clean), lint (no new errors in the touched files), and the test suite - all 18 pass (URL-parsing edge cases, fetch success/failure/missing-token, ResourceDetailPage integration). Nice, clean implementation.
